bin/python-stripped-env

```diff
@@ -30,8 +30,28 @@
 # -u  Treat unset variables as an error when substituting.
 set -e -u
 
-# actual python binary that we'll call
-PYTHON=/usr/bin/python
+# default python interpreter
```

Comment:
i'm trying to understand what this is fixing, but i can't get my had wrapped around it

Reply:
ok, nvm. i'm not a big fan of this tbh. i don't like the fact that i can eat arguments, making it a bad wrapper.
better solutions however...
in what kind of an environment do you run into this?

Reply:
It does not really depend on any specific environment. The problem is that `vsc-install` is breaking any scripts that can only be executed in Python 3. Scripts with a shebang such as `#!/usr/bin/env python3` are replaced with `#!/usr/bin/python-stripped-env` which is hardcoded to `/usr/bin/python`. Therefore, if the Python 3 interpreter is called `python3`, as it usually is, things break.

I propose to solve it by allowing to pass an argument to `python-stripped-env` defining the target interpreter. In a similar way to how `/usr/bin/env` works. This PR does not change the default behaviour of `python-stripped-env`.

Reply:
https://github.com/hpcugent/vsc-install/pull/175/commits/7f647340b125200e4a75adcb9f658722c0a18018 tightens the names considered as possible python interpreters to reduces the chances of eating arguments by mistake. Now it only accepts the basic `python`, `python2`, `python3` and those followed by any dot version.